File: OOAD_HW1/Q4/elephant.java
package Q4;

public class elephant extends pachyderm {
	
	// Elephant's name
	private String name;
	
	// Default Constructor, don't use.
	public elephant() {
		super("None");
		this.name = "None";
	}
	
	// Constructor for setting the hippo's name, and it's super-class' name.
	public elephant(String name) {
		super(name);
		this.name = name;
	}
}

File: OOAD_HW2/Part B/Elephant.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <iomanip>

using namespace std;

class Elephant: public Pachyderm
{
    protected:
        string name;
        string type;
        //NoiseBehavior *noise;

    public:
        Elephant(string aName, string aType);
};

// Behavior handled by interface via super class.
Elephant::Elephant(string aName, string aType): Pachyderm(aName, aType)
{
    name = aName;
    type = aType;
}



